src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -456,5 +456,330 @@ public static void OverrideJsonIgnorePropertyUsingJsonPropertyNameReversed()
+            string json = @"{""MyString"":""Random"",""MyDateTime"":""2020-03-23"",""MyInt"":4}";
+
+            object obj = JsonSerializer.Deserialize(json, type);
+            Assert.Equal("Random", (string)type.GetProperty("MyString").GetValue(obj));
+            Assert.Equal(default, (DateTime)type.GetProperty("MyDateTime").GetValue(obj));
+            Assert.Equal(4, (int)type.GetProperty("MyInt").GetValue(obj));
+
+            string serialized = JsonSerializer.Serialize(obj);
+            Assert.Contains(@"""MyString"":""Random""", serialized);
+            Assert.Contains(@"""MyInt"":4", serialized);
+            Assert.DoesNotContain(@"""MyDateTime"":", serialized);
+        }
+
+        private class ClassWithProperty_IgnoreConditionAlways
+        {
+            public string MyString { get; set; }
+            [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+            public DateTime MyDateTime { get; set; }
+            public int MyInt { get; set; }
+        }
+
+        private class ClassWithProperty_IgnoreConditionAlways_Ctor
+        {
+            public string MyString { get; set; }
+            [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+            public DateTime MyDateTime { get; }
+            public int MyInt { get; }
+
+            public ClassWithProperty_IgnoreConditionAlways_Ctor(DateTime myDateTime, int myInt)
+            {
+                MyDateTime = myDateTime;
+                MyInt = myInt;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonIgnoreConditionWhenNull_ClassProperty_TestData))]
+        public static void JsonIgnoreConditionWhenNull_ClassProperty(Type type, JsonSerializerOptions options)
+        {
+            // Property shouldn't be ignored if it isn't null.
+            string json = @"{""Int1"":1,""MyString"":""Random"",""Int2"":2}";
+
+            object obj = JsonSerializer.Deserialize(json, type, options);
+            Assert.Equal(1, (int)type.GetProperty("Int1").GetValue(obj));
+            Assert.Equal("Random", (string)type.GetProperty("MyString").GetValue(obj));
+            Assert.Equal(2, (int)type.GetProperty("Int2").GetValue(obj));
+
+            string serialized = JsonSerializer.Serialize(obj, options);
+            Assert.Contains(@"""Int1"":1", serialized);
+            Assert.Contains(@"""MyString"":""Random""", serialized);
+            Assert.Contains(@"""Int2"":2", serialized);
+
+            // Property should be ignored when null.
+            json = @"{""Int1"":1,""MyString"":null,""Int2"":2}";
+
+            obj = JsonSerializer.Deserialize(json, type, options);
+            Assert.Equal(1, (int)type.GetProperty("Int1").GetValue(obj));
+            Assert.Equal("DefaultString", (string)type.GetProperty("MyString").GetValue(obj));
+            Assert.Equal(2, (int)type.GetProperty("Int2").GetValue(obj));
+
+            // Set property to be ignored to null.
+            type.GetProperty("MyString").SetValue(obj, null);
```

Comment:
Wouldn't this affect other tests that re-use the class? If there is none, disregard this comment.

Reply:
This sets the value for the `obj` instance only, so it's only relevant in the current scope.

Reply:
Oh you are right.